src/pages/NewChatPage.js

```diff
@@ -167,6 +169,13 @@ function NewChatPage({betas, isGroupChat, personalDetails, reports, translate})
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [reports, personalDetails, searchTerm]);
 
+    // When search term updates we will fetch any reports
+    const setSearchTermAndSearchInServer = useCallback((text = '') => {
+        if (text.length) {
+            Report.searchInServer(text);
+        }
+        setSearchTerm(text);
+    }, []);
```

Comment:
NAB: Wondering if we should have kept `onChangeText={setSearchTerm}` and use a `useEffect`

```
useEffect(() => {
        if (searchTerm.length) {
            Report.searchInServer(searchTerm);
        }
}, [searchTerm]);
```
To do trigger the server search. I think I like the `useEffect` pattern more, but NAB!

Reply:
Hehe, I see this is exactly the opposite of @hoangzinh 's preference: https://github.com/Expensify/App/pull/27819#discussion_r1340994924

Reply:
For me it makes sense to use `useEffect` because I see it as a side effect of the the `searchTerm`  changing. Also a benefit of `useEffect` is that you could cancel the debounced call if the components is unmounted... like:

```js
useEffect(() => {
        if (searchTerm.length) {
            Report.searchInServer(searchTerm);
        }
        return () => Report.searchInServer.cancel(); // https://stackoverflow.com/a/41286377
}, [searchTerm]);
```

but maybe we don't care about canceling

* the cancel example above is not working because you have to call `.cancel` on the debounced function and `Report.searchInServer` is not the debounced function... but it is just to show the idea.

Reply:
Personally, I don't think we need to optimize the case of canceling the debounced function - but maybe would if it was directly calling a `setState()` method after the component has unmounted. 

In this case, the chats will load with or without the page and it won't really have any downsides.